core/src/smartcontracts/wasm.rs

```diff
@@ -1348,17 +1438,18 @@ impl<'wrld> import_traits::SetPermissionTokenSchema<state::validator::Migrate<'w
 }
 
 /// `Runtime` builder
-#[derive(Default)]
 pub struct RuntimeBuilder<S> {
+    query_handle: LiveQueryStoreHandle,
     engine: Option<Engine>,
     config: Option<Configuration>,
     linker: Option<Linker<S>>,
 }
 
 impl<S> RuntimeBuilder<S> {
     /// Creates a new [`RuntimeBuilder`]
-    pub fn new() -> Self {
+    pub fn new(query_handle: LiveQueryStoreHandle) -> Self {
```

Comment:
why is query store given from the outside? Does this mean that query store is shared between different smart contracts?

Reply:
This is just a handle to a one service running on host on the background. So any smart contract can talk to it, yes

Reply:
then you should make sure that you have deleted all live queries after smart contract exits (whether successfully or with error)

Reply:
That's a nice idea. Don't you mind if I implement it in another PR?

Reply:
- Created #3982